chalice/app.py

```diff
@@ -232,6 +232,11 @@ def get_access_control_headers(self):
 
         return headers
 
+    def __eq__(self, other):
```

Comment:
Yeah I think I want to stick with equality for now. Mainly because it may be a bit of pain to pass the same object around for all of the routes when I should just be able to create two objects that are exactly the same. I think it is important that this is documented at least so users are aware of this behavior when designing their API.